### PR TITLE
scheduler(ticdc): optimize owner memory allocation

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: ref #7720 

### What is changed and how it works?

* pkg/spanz(ticdc): increase btree degree to speed up Ascend
* *(ticdc): cache []tablepb.Span
* pkg/spanz(ticdc): do not let variables move to heap 
* scheduler(ticdc): do not let span move to heap

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

 Replicate 72k tables. Left master branch, right this PR.
* `alloc_objects` are reduced from 46% to less than 1% 
* `cpu` are almost the same, from 3.44% to 3.19%
image
image

##### Will it cause performance regression or break compatibility?

No.

##### Do you need to update user documentation, design documentation or monitoring documentation?

No.

### Release note 

```release-note
None
```

### CODE REVIEW

File: pkg/spanz/map.go [hunk 1]
```diff
@@ -109,52 +117,65 @@ func (m *Map[T]) AscendRange(start, end tablepb.Span, iterator ItemIterator[T])
...
 // FindHoles returns an array of Span that are not covered in the range
 // [start, end).
-// Note: Table ID is not set in returned holes.
-func (m *Map[T]) FindHoles(start, end tablepb.Span) (coveredSpans, holes []tablepb.Span) {
+// Note:
+// * Table ID is not set in returned holes.
+// * Returned slice is read only and will be changed on next FindHoles.
+func (m *Map[T]) FindHoles(start, end tablepb.Span) ([]tablepb.Span, []tablepb.Span) {
 	if bytes.Compare(start.StartKey, end.StartKey) >= 0 {
 		log.Panic("start muse be larger than end",
```
Comment (zhaoxinyu) [inline pkg/spanz/map.go:125]:
```suggestion
		log.Panic("start must be larger than end",
```

File: pkg/spanz/map.go [hunk 2]
```diff
@@ -36,12 +37,19 @@ func lessSpanItem[T any](a, b spanItem[T]) bool {
...
+	cache *struct {
+		coveredSpans, holes []tablepb.Span
+		lastGC              time.Time
+	}
 }
 
 // NewMap returns a new SpanMap.
 func NewMap[T any]() *Map[T] {
+	// Map is read heavy map, many Ascend.
+	const defaultDegree = 256
```
Comment (zhaoxinyu) [inline pkg/spanz/map.go:50]: Maybe it's better to move this constant variable out of `NewMap()`

File: pkg/spanz/map.go [hunk 3]
```diff
@@ -109,52 +117,65 @@ func (m *Map[T]) AscendRange(start, end tablepb.Span, iterator ItemIterator[T])
...
+			zap.String("end", end.String()))
+	}
+	if m.cache == nil || time.Since(m.cache.lastGC) > time.Minute {
+		m.cache = &struct {
+			coveredSpans []tablepb.Span
+			holes        []tablepb.Span
+			lastGC       time.Time
+		}{lastGC: time.Now()}
 	}
+	m.cache.coveredSpans = m.cache.coveredSpans[:0]
```
Comment (3AceShowHand) [inline pkg/spanz/map.go:136]: How about add a new method named `CleanCache` to replace this 2 lines.

**Review by 3AceShowHand — APPROVED** [review]

File: pkg/spanz/map.go (hunk 2 above)
Comment (overvenus) [reply pkg/spanz/map.go:50]: I file a task in https://github.com/pingcap/tiflow/issues/7720, the btree map will only be used in scheduler package which has to deal with a larger number of tables and lots of Ascend. I will address your comment in the task.

**Review by zhaoxinyu — APPROVED** [review]

**Pull request merged**